samples/drivers/espi/src/main.c

```diff
@@ -309,6 +320,119 @@ int espi_handshake(void)
 	}
 
 	LOG_INF("2nd phase completed");
+
+	ret = wait_for_vwire(espi_dev, ESPI_VWIRE_SIGNAL_PLTRST,
+			     CONFIG_ESPI_VIRTUAL_WIRE_TIMEOUT, 1);
+	if (ret) {
+		LOG_WRN("PLT_RST Timeout");
+		return ret;
+	}
+
+	LOG_INF("3rd phase completed");
+
+	return 0;
+}
+
+int read_test_block(u8_t *buf, u32_t start_flash_adr, u16_t block_len)
+{
+	u8_t i = 0;
+	u32_t flash_addr = start_flash_adr;
+	u16_t transactions = block_len/MAX_FLASH_REQUEST;
+	int ret = 0;
+	struct espi_flash_packet pckt;
+
+	for (i = 0; i < transactions; i++) {
+		pckt.buf = buf;
+		pckt.flash_addr = flash_addr;
+		pckt.len = MAX_FLASH_REQUEST;
+
+		ret = espi_read_flash(espi_dev, pckt);
+		if (ret) {
+			LOG_WRN("espi_read_flash failed: %d", ret);
+			return ret;
+		}
+
+		buf += MAX_FLASH_REQUEST;
+		flash_addr += MAX_FLASH_REQUEST;
+	}
+
+	LOG_INF("%d transactions completed\n", transactions);
+	return 0;
+}
+
+int write_test_block(u8_t *buf, u32_t start_flash_adr, u16_t block_len)
+{
+	u8_t i = 0;
+	u32_t flash_addr = start_flash_adr;
+	u16_t transactions = block_len/MAX_FLASH_REQUEST;
+	int ret = 0;
+	struct espi_flash_packet pckt;
+
+	/* Split operation in multiple MAX_FLASH_REQ transactions */
+	for (i = 0; i < transactions; i++) {
+		pckt.buf = buf;
+		pckt.flash_addr = flash_addr;
+		pckt.len = MAX_FLASH_REQUEST;
+
+		ret = espi_write_flash(espi_dev, pckt);
+		if (ret) {
+			LOG_WRN("espi_write_flash failed: %d", ret);
+			return ret;
+		}
+
+		buf += MAX_FLASH_REQUEST;
+		flash_addr += MAX_FLASH_REQUEST;
+	}
+
+	LOG_INF("%d transactions completed\n", transactions);
+	return 0;
+}
+
+static int test_espi_flash(u32_t start_flash_addr, u8_t blocks)
+{
+	u8_t i;
+	u8_t pattern;
+	u32_t flash_addr;
+	int ret = 0;
+
+	LOG_INF("Test eSPI write flash\n");
+	flash_addr = start_flash_addr;
+	pattern = 0x99;
+	for (i = 0; i <= blocks; i++) {
+		memset(flash_write_buf, pattern++, sizeof(flash_write_buf));
+		ret = write_test_block(flash_write_buf, flash_addr,
+				       sizeof(flash_write_buf));
+		if (ret) {
+			LOG_WRN("Failed to write to eSPI\n");
+			return ret;
+		}
+
+		flash_addr += sizeof(flash_write_buf);
+	}
+
+	LOG_INF("Test eSPI read flash\n");
+	flash_addr = start_flash_addr;
+	pattern = 0x99;
+	for (i = 0; i <= blocks; i++) {
+		/* Clear last read content */
+		memset(flash_read_buf, 0, sizeof(flash_read_buf));
+		ret = read_test_block(flash_read_buf, flash_addr,
+				      sizeof(flash_read_buf));
+		if (ret) {
+			LOG_WRN("Failed to read from eSPI\n");
+			return ret;
+		}
+
+		/* Simple check */
```

Comment:
I think a full buffer check would be better. It can catch issues when buffer is being passed around.

Reply:
Yes, updated